File: agent-core/src/cache.ts
interface CacheEntry<T> {
  value: T;
  timestamp: number;
}

const cache = new Map<string, CacheEntry<any>>();

export function get<T>(key: string, ttl: number = 3600): T | undefined {
  const entry = cache.get(key);
  if (!entry) return undefined;
  
  if (Date.now() - entry.timestamp > ttl * 1000) {
    cache.delete(key);
    return undefined;
  }
  
  return entry.value;
}

export function set<T>(key: string, value: T): void {
  cache.set(key, {
    value,
    timestamp: Date.now()
  });
}

export function clear(): void {
  cache.clear();
}

export function size(): number {
  return cache.size;
}

export function cleanup(maxAge: number = 3600): number {
  const now = Date.now();
  let removed = 0;
  
  for (const [key, entry] of cache.entries()) {
    if (now - entry.timestamp > maxAge * 1000) {
      cache.delete(key);
      removed++;
    }
  }
  
  return removed;
}

File: backend/core/cache.py
import time
from typing import Any, Optional

class Cache:
    """Simple in-memory cache with TTL support."""
    
    def __init__(self, ttl: int = 3600):
        self.ttl = ttl
        self.store = {}
    
    def get(self, key: str) -> Optional[Any]:
        """Get value from cache if not expired."""
        entry = self.store.get(key)
        if not entry:
            return None
        
        if time.time() - entry['timestamp'] > self.ttl:
            self.store.pop(key, None)
            return None
            
        return entry['value']
    
    def set(self, key: str, value: Any) -> None:
        """Set value in cache with current timestamp."""
        self.store[key] = {
            'value': value,
            'timestamp': time.time()
        }
    
    def clear(self) -> None:
        """Clear all cached entries."""
        self.store.clear()
    
    def size(self) -> int:
        """Get number of cached entries."""
        return len(self.store)